src/main/kotlin/no/nav/helse/oppslag/infotrygd/InfotrygdSakClient.kt

```diff
@@ -1,17 +1,32 @@
 package no.nav.helse.oppslag.infotrygd
 
 import arrow.core.Try
+import arrow.core.failure
 import no.nav.helse.common.toXmlGregorianCalendar
+import no.nav.helse.oppslag.infotrygdberegningsgrunnlag.BaseneErUtilgjengeligeException
 import no.nav.tjeneste.virksomhet.infotrygdsak.v1.binding.InfotrygdSakV1
 import no.nav.tjeneste.virksomhet.infotrygdsak.v1.informasjon.Periode
 import no.nav.tjeneste.virksomhet.infotrygdsak.v1.meldinger.FinnSakListeRequest
 import java.time.LocalDate
+import javax.xml.ws.soap.SOAPFaultException
 
 class InfotrygdSakClient(private val port: InfotrygdSakV1) {
 
+    companion object {
+        private val baseneErUtilgjengelige = "Basene i Infotrygd er ikke tilgjengelige"
+    }
+
     fun finnSakListe(fødselsnummer: String, fom: LocalDate, tom: LocalDate?) =
             Try {
                 port.finnSakListe(finnSakListeRequest(fødselsnummer, fom, tom))
+            }.let {
+                if (it is Try.Failure && it.exception is SOAPFaultException) {
+                    if ((it.exception as SOAPFaultException).message == baseneErUtilgjengelige) {
+                        return@let BaseneErUtilgjengeligeException(it.exception as SOAPFaultException)
+                                .failure()
+                    }
+                }
+                it
```

Comment:
Bruke `fold()` istedenfor `it is Try.Failure`?

Reply:
`fold()` mapper både success og failure til ny (samme) type, og jeg mapper egentlig bare over feil-typen. Either har `mapLeft`, men Try ser ut til å mangle denne :(

Reply:
`Try.toEither() da`, synes bare `return @` er grisete, får `goto`-assosiasjoner

Reply:
enig, men `return` er heldigvis bare ut av `let`, som er samme kodeblokk. Alternativet er å ha en else-blokk etter if-en:
```
} else {
    it
}
```

Klient-klassene returnerer en `Try`, så blir det mappet om til `Either` i de ulike tjenestene. Rationale mitt her er at jeg ser på klient-klassene som noe som er på "utsiden", og har ikke noe kjennskap til domene-feilmeldingene til `Feilårsak`.